packages/build/src/core/config.js

```diff
@@ -152,6 +152,7 @@ export const saveUpdatedConfig = async function ({
   buildDir,
   repositoryRoot,
   configPath = `${repositoryRoot}/netlify.toml`,
+  outputConfigPath = configPath,
```

Comment:
This is what ensures that the current deploy site step continues to work in the same way, as the output config path will just default to the input config path.